src/dune_pkg/lock_dir.ml

```diff
@@ -408,7 +408,19 @@ module Write_disk = struct
             Dune_lang.Ast.add_loc ~loc:Loc.none sexp |> Dune_sexp.Cst.concrete)
         in
         let pp = Dune_lang.Format.pp_top_sexps ~version:(3, 11) cst in
-        Format.asprintf "%a" Pp.to_fmt pp |> Io.write_file path)
+        Format.asprintf "%a" Pp.to_fmt pp |> Io.write_file path;
+        Package_name.Map.iteri files ~f:(fun package_name files ->
+          let files_dir =
+            Path.relative lock_dir_path (Package_name.to_string package_name ^ ".files")
+          in
+          if List.is_non_empty files then Path.mkdir_p files_dir;
+          List.iter files ~f:(fun (original_file, local_file) ->
+            let dst = Path.append_local files_dir local_file in
+            Path.mkdir_p (Path.parent_exn dst);
```

Comment:
This might be a bit wasteful, however creating all the subdirectories is a bit annoying otherwise.

Reply:
No, it's totally fine.